File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int i=0;
        double sum=0;
        double maxavg=Integer.MIN_VALUE;
        for(int j=0;j<nums.length;j++){
            sum+=nums[j];
            if(j-i+1>k){
                sum-=nums[i];
                i++;
            }
            if(j-i+1==k){
                double avg=sum/k;
                maxavg=Math.max(avg,maxavg);
            }
            
        }
        return maxavg;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution(object):
    def findMaxAverage(self, nums, k):

        curr_sum = 0
        i = j = 0
        max_sum = float('-inf')

        while j < len(nums):

            curr_sum += nums[j]

            if j - i + 1 < k:
                j += 1

            elif j - i + 1 == k:

                max_sum = max(max_sum, curr_sum)

                curr_sum -= nums[i]
                i += 1
                j += 1

        return max_sum / float(k)